File: day3/two_sum.cpp
#include<iostream>
#include<vector>
#include<map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> indexMap; 
        int complement = 0;
        int i = 0;

        for(; i < nums.size(); ++i){
            complement = target - nums[i];
            if(indexMap.count(complement) == 1)
                break;
            else
                indexMap[nums[i]] = i;
        }

        return {i, indexMap[complement]};
    }
};

File: solutions/day_003/two_sum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    //get the length of the array
    int numsLength = nums.length;

    for (int i = 0; i < numsLength - 1; i++) {
      //loop through the array and check if the sum of the current element and the next element is equal to the target
      for (int j = i + 1; j < numsLength; j++) {
        if (nums[i] + nums[j] == target) {
          return [i, j];
        }
      }
    }

    return [];
  }
}
